internal/client/config.go

```diff
@@ -0,0 +1,253 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package client
+
+import (
+	"crypto/sha256"
+	"crypto/tls"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+
+	tfe "github.com/hashicorp/go-tfe"
+	"github.com/hashicorp/hcl"
+	"github.com/hashicorp/terraform-provider-tfe/internal/logging"
+	providerVersion "github.com/hashicorp/terraform-provider-tfe/version"
+	svchost "github.com/hashicorp/terraform-svchost"
+	"github.com/hashicorp/terraform-svchost/auth"
+	"github.com/hashicorp/terraform-svchost/disco"
+)
+
+var (
+	// TFEUserAgent is the user agent string sent with all requests made by the provider
+	TFEUserAgent = fmt.Sprintf("terraform-provider-tfe/%s", providerVersion.ProviderVersion)
+)
+
+type CredentialsMap map[string]map[string]interface{}
+
+// CLIHostConfig is the structure of the configuration for the Terraform CLI.
+type CLIHostConfig struct {
+	Hosts       map[string]*ConfigHost `hcl:"host"`
+	Credentials CredentialsMap         `hcl:"credentials"`
+}
+
+// ConfigHost is the structure of the "host" nested block within the CLI
+// configuration, which can be used to override the default service host
+// discovery behavior for a particular hostname.
+type ConfigHost struct {
+	Services map[string]interface{} `hcl:"services"`
+}
+
+// ClientConfiguration is the refined information needed to configure a tfe.Client
+type ClientConfiguration struct {
+	Services   *disco.Disco
+	HTTPClient *http.Client
+	TFEHost    svchost.Hostname
+	Token      string
+	Insecure   bool
+}
+
+// Key returns a string that is comparable to other ClientConfiguration values
+func (c ClientConfiguration) Key() string {
+	return fmt.Sprintf("%x %s/%v", sha256.New().Sum([]byte(c.Token)), c.TFEHost, c.Insecure)
```

Comment:
Oh, ha, okay — I don't know if there's any way you could manage to configure an insecure and a secure client for the same hostname, but I can definitely see why we'd want them in separate slots if someone managed it.

Reply:
You can add it to the provider config as `ssl_skip_verify` So you _could_ be aliasing the provider with an insecure for some odd reason.